packages/react-components/babel-preset-global-context/src/transformPlugin.ts

```diff
@@ -0,0 +1,259 @@
+  contextSelectorLocalName?: string;
+};
+
+interface PackageJSON {
+  name: string;
+  version: string;
+}
+
+/**
+ * Checks that passed callee imports react context or context selector
+ */
+function isCreateContextCallee(path: NodePath<t.Expression | t.V8IntrinsicIdentifier>): path is NodePath<t.Identifier> {
+  if (!path.isIdentifier) {
+    return false;
+  }
+
+  return (
+    path.referencesImport(REACT_PACKAGE, CREATE_CONTEXT_CALL) ||
+    path.referencesImport(CONTEXT_SELECTOR_PACKAGE, CREATE_CONTEXT_CALL)
+  );
+}
+
+/**
+ *
+ * @param packageName - name of the package to import
+ * @param functionName - name of the createContext function
+ * @returns a new import expression for global context
+ *
+ * @example import { createContext as __createGlobalContext } from '@global-context/react'
+ */
+function createGlobalContextImportDeclaration(packageName: string, isContextSelector: boolean) {
+  const origFunction = isContextSelector ? 'createContextSelector' : 'createContext';
+  const newFunction = isContextSelector ? GLOBAL_CONTEXT_SELECTOR_ALIAS : GLOBAL_CONTEXT_ALIAS;
+  return types.importDeclaration(
+    [types.importSpecifier(types.identifier(newFunction), types.identifier(origFunction))],
+    types.stringLiteral(packageName),
+  );
+}
+
+/**
+ * @returns - An expression that creates a global context
+ *
+ * @example const MyContext = __createGlobalContext();
+ */
+function createGlobalContextCallExpression(options: {
+  expressionPath: NodePath<t.CallExpression>;
+  packageJson: PackageJSON;
+  packageJsonPath: string;
+  filePath: string;
+  functionName: string;
+}) {
+  const { expressionPath, packageJson, packageJsonPath, filePath, functionName } = options;
+
+  const args = expressionPath.get('arguments').map(arg => arg.node);
+  if (!expressionPath.parentPath.isVariableDeclarator()) {
+    return expressionPath.node;
+  }
+
+  // Use the relative path from package.json because the same package
+  // can be installed under different paths in node_modules if they are duplicated
+  const relativePath = relative(packageJsonPath, filePath);
+  const id = expressionPath.parentPath.get('id') as NodePath<babel.types.Identifier>;
+  return types.callExpression(types.identifier(functionName), [
+    ...args,
+    types.stringLiteral(hash(`${relativePath}@${id.node.name}`)),
+    types.stringLiteral(packageJson.name),
+    types.stringLiteral(packageJson.version),
+  ]);
+}
+
+/**
+ * Checks if import statement import createContext() from React
+ */
+function hasReactImport(path: NodePath<babel.types.ImportDeclaration>): boolean {
+  return path.node.source.value === REACT_PACKAGE;
+}
+
+/**
+ * Checks if import statement import createContext() from @fluentui/react-context-selector
+ */
+function hasContextSelectorImport(path: NodePath<babel.types.ImportDeclaration>): boolean {
+  return path.node.source.value === CONTEXT_SELECTOR_PACKAGE;
+}
+
+export const transformPlugin = declare<{}, PluginObj<BabelPluginState>>(api => {
```

Comment:
**chore:** manual generic instantiation is usually a code smell and points that author API (that declare function) might be using generic in invalid way. 

This case seems to be indeed an code smell   - using `{}` former bottom type manually for no implementation purpose (because constraint from declare definition)

we can annotate return value of our callback and all is fixed

```suggestion
export const transformPlugin = declare((api): PluginObj<BabelPluginState> => {
```

Reply:
nice, I have to remember how powerful inference is in the future